src/main/user-impl/java/com/mysql/cj/jdbc/ha/ConnectionProxy.java

```diff
@@ -225,7 +226,13 @@ protected void initLogger(ConnectionUrl connUrl) {
 
   protected void initSettings(ConnectionUrl connectionUrl) throws SQLException {
     try {
-      this.connProps.initializeProperties(connectionUrl.getConnectionArgumentsAsProperties());
+      Properties props = new Properties();
+      for (Map.Entry<String, String> entry : connectionUrl.getMainHost().getHostProperties().entrySet()) {
+        if (entry.getValue() != null) {
+          props.put(entry.getKey(), entry.getValue());
+        }
+      }
+      this.connProps.initializeProperties(props);
```

Comment:
```suggestion
      final Properties props = new Properties();
      props.putAll(connectionUrl.getMainHost().getHostProperties());
      this.connProps.initializeProperties(props);
```

Reply:
This will cause a null pointer exception if one of the properties is null.